Create net-pricing in distributor (#13)
MewsSystems/gitbook-distributor-guide: distributor-api-v1/net-pricing.md

# Pricing

**Net price** is an amount without taxes.  
**Gross price** is a net price + taxes.

> In order to use net pricing in distributor, the API returns the [Amount](net-pricing.md#Amount) object as part of responses.

Reviewer: ```suggestion
In order to use pricing in the distributor, the API returns the [Amount](#amount) object as part of responses.
```

the manual is not just for net pricing, but for both net and gross pricing (as per your title).
this is not how links in gitbook work imo

Reviewer: do not use newlines inside paragraph, it is either new paragraph or next sentence in the paragraph. Using newlines to style a document is wrong.

[Amount](net-pricing.md#Amount) represents a structure that holds gross price, net price and tax values.

The Amount has following structure:
```javascript
Currency:
{

>     "GrossValue": 100.00,

Reviewer 2: Bad indentation here

	"NetValue": 93.46,
	"TaxValues": [
		{
			"TaxRateCode": "DE-R",
			"Value": 6.54
		}
	]
}
```

### Amount <a id="amount"></a>

> |  | Property | Description |

Reviewer: ```suggestion
| Property | Type | Description |
```

| :--- | :--- | :--- |
| `GrossValue` | Number | Gross value  |
| `NetValue` | Number | Net value |

> | `TaxValues` | [TaxValue](net-pricing.md.md#taxValue) | Tax value for the net value amount |

Reviewer: ```suggestion
| `TaxValues` | Collection of [TaxValue](#taxValue)s | Tax values for the net value amount |
```

> ### TaxValue <a id="taxValue"></a>

Reviewer: ```suggestion
### TaxValue
```

should not be needed in the headings

|  | Property | Description |
| :--- | :--- | :--- |
| `TaxRateCode` | string | Unique identifier of the rate. |
| `Value` | Number | Amount of tax |

In API response it is represented as this:
```javascript
{
    "TotalAmount": {
        "USD": {
		"GrossValue": 100.00,
		"NetValue": 93.46,
		"TaxValues": [
			{
				"TaxRateCode": "DE-R",
				"Value": 6.54
			}
		]
	},
	"GBP": {
		"GrossValue": 90.00,
		"NetValue": 83.46,
		"TaxValues": [
			{
				"TaxRateCode": "DE-R",
				"Value": 6.54
			}
		]
	}
    }
}
```

> This "TotalAmount" object will be sent from the API and it will contain values for every currency which is represented by [Amount](net-pricing.md#Amount). TotalAmount is dictionary with Currency as a key and [Amount](net-pricing.md#Amount) as value.

Reviewer: new line around the code block (check whole file)

# Pricing

**Net price** is an amount that does not have taxes included.  
**Gross price** is an amount that already includes a net price + taxes. Examples of this are below in the responses to API calls.

> In order to use net pricing in distributor, the "Amount" object, that is returned from the API, has to be used. In every response that has something to do with pricing, the API returns two objects:

Reviewer: document only the state we are keeping, do not mention the costs, nor in examples.

1. Old object (Cost) that represents currency and value. This is an old representation and should not be used since it is going to be deprecated by the end of June. After that, API will stop sending it.
2. New object (Amount) that represents a structure that holds both gross and net price. This object is sent always and is the one to be used. It is going to be the only object sent therefore it is mandatory.

> Cost object:

Reviewer: we want to remove them (as per your 1), we should not document them.

```javascript
"GTQ": 540.49
```

In API response it is represented as this:
```javascript
"Total": 
{
  "GTQ": 540.49,
  "RUB": 4433.62
}
```

The new "Amount" structure. This one is the one to be used and supported. 

The object itself has following structure:
```javascript
"GTQ":
{
  "GrossValue": 540.49,
  "NetValue": 540.49,
  "TaxValues": [
    {
      "TaxRateCode": "CZ-Z",
      "Value": 0.00
    }
  ]
}
```

In API response it is represented as this:
```javascript
{
    "TotalAmount": {
        "GTQ": {
            "GrossValue": 540.49,
            "NetValue": 540.49,
            "TaxValues": [
                {
                    "TaxRateCode": "CZ-Z",
                    "Value": 0.00
                }
            ]
        },
        "RUB": {
            "GrossValue": 4433.62,
            "NetValue": 4433.62,
            "TaxValues": [
                {
                    "TaxRateCode": "CZ-Z",
                    "Value": 0.00
                }
            ]
        }
    }
}
```
This "TotalAmount" object will be sent from the API together with the "Total" object. Because the one that should be used has "Amount" in its name, the "Total" object should be ignored.

> Be aware that the API responses can be quite large - more about that later.  

Reviewer 2: Maybe this doesn't have to be mentioned - for most properties, they will only have a few rates - hence the response should not be that big. Moreover, since the duplicate object will ultimately be removed, this is a temporary effect.

Following is an example request for pricing

> Request:

Reviewer 2: Mention endpoint name.
